src/utils/toggle-bookmark.ts

```diff
@@ -1,16 +1,17 @@
 import { userService } from '~/services/user-service'
-import useAxios, { UseAxiosProps } from '~/hooks/use-axios'
+import useMutation from '~/hooks/use-mutation'
+import { UseAxiosProps } from '~/hooks/use-axios'
 
 export const useToggleBookmark = (
   userId: string,
   onResponse: UseAxiosProps<string[], string>['onResponse'],
   onResponseError: UseAxiosProps<string[], string>['onResponseError']
 ) => {
-  const { fetchData: toggleBookmark } = useAxios<string[], string>({
-    service: (offerID: string) => userService.toggleBookmark(userId, offerID),
-    fetchOnMount: false,
-    onResponse,
-    onResponseError
+  const { mutate: toggleBookmark } = useMutation({
+    mutationFn: (offerID: string) =>
+      userService.toggleBookmark(userId, offerID),
+    onSuccess: (response) => onResponse?.(response),
+    onError: (error) => onResponseError?.(error)
```

Comment:
Does it work like that?
```suggestion
    onSuccess: onResponse,
    onError: onResponseError
```

Reply:
Yes, it works fine.

Reply:
Changed to your suggestion, looks cleaner, thanks.